deploy/docker/docker-entrypoint.sh

```diff
@@ -101,103 +101,114 @@ fi
+            if [[ -n "$VAR_VALUE" ]]; then
+                VAR_TEMPLATE_NAME="$(echo "$VAR_NAME" | sed -r -e 's/\.[0-9]+/.[0-9]+/g' -e 's/[0-9]+\./[0-9]+./g' -e 's/\./\\./g')"
+                if grep -qE "$VAR_TEMPLATE_NAME" "$CONFIG"; then
+                    echo_value
+                    sed -i '$a'\\ "$CONFIG"
+                    echo "$VAR_NAME = $VAR_VALUE" >>"$CONFIG"
+                fi
             fi
         fi
         # Config in plugins/*
-        for CONFIG_PLUGINS_FILE in $(ls $CONFIG_PLUGINS); do
-            if [[ ! -z "$(cat $CONFIG_PLUGINS/$CONFIG_PLUGINS_FILE |grep -E "^(^|^#*|^#*\s*)$VAR_NAME")" ]]; then
+        for CONFIG_PLUGINS_FILE in "$CONFIG_PLUGINS"/*; do
+            if [[ -n "$(perl -sne 'print if /^[#\s]*\Q$var_name\E\s*=/' -- -var_name="$VAR_NAME" "$CONFIG_PLUGINS_FILE")" ]]; then
                 echo_value
-                if [[ -z "$(eval echo \$$VAR_FULL_NAME)" ]]; then
-                    echo "$(sed -r "s/(^\s*)($VAR_NAME\s*=\s*.*)/#\2/g" $CONFIG_PLUGINS/$CONFIG_PLUGINS_FILE)" > $CONFIG_PLUGINS/$CONFIG_PLUGINS_FILE
+                if [[ -z "$VAR_VALUE" ]]; then
+                    perl -i -spe 's/^[#\s]*(\Q$var_name\E)\s*=\s*(.*)/# \1 = \2/' -- -var_name="$VAR_NAME" "$CONFIG_PLUGINS_FILE"
                 else
-                    echo "$(sed -r "s/(^#*\s*)($VAR_NAME)\s*=\s*(.*)/\2 = $(eval echo \$$VAR_FULL_NAME|sed -e 's/\//\\\//g')/g" $CONFIG_PLUGINS/$CONFIG_PLUGINS_FILE)" > $CONFIG_PLUGINS/$CONFIG_PLUGINS_FILE
+                    perl -i -spe 's/^[#\s]*(\Q$var_name\E)\s*=\s*(.*)/\1 = $var_value/' -- -var_name="$VAR_NAME" -var_value="$VAR_VALUE" "$CONFIG_PLUGINS_FILE"
                 fi
             # Check if config has a numbering system, but no existing configuration line in file
-            elif [[ ! -z "$(echo "$VAR_NAME" | grep '\(\.[0-9]\|[0-9]\.\)')" ]]; then
-                if [[ ! -z "$(eval echo \$$VAR_FULL_NAME)" ]]; then
-                    VAR_TEMPLATE_NAME="$(echo "$VAR_NAME" | sed -r -e 's|[.]|\\.|g' -e 's|\\.[0-9]+|\\.[0-9]|' -e 's|[0-9]+\\.|[0-9]\\.|')"
-                    if [[ ! -z "$(grep "$VAR_TEMPLATE_NAME" "$CONFIG_PLUGINS/$CONFIG_PLUGINS_FILE")" ]]; then
+            elif echo "$VAR_NAME" | grep -qE '\.\d+|\d+\.'; then
+                if [[ -n "$VAR_VALUE" ]]; then
+                    VAR_TEMPLATE_NAME="$(echo "$VAR_NAME" | sed -r -e 's/\.[0-9]+/.[0-9]+/g' -e 's/[0-9]+\./[0-9]+./g' -e 's/\./\\./g')"
+                    if grep -qE "$VAR_TEMPLATE_NAME" "$CONFIG_PLUGINS_FILE"; then
                         echo_value
-                        echo "$VAR_NAME = $(eval echo \$$VAR_FULL_NAME)" >> $CONFIG_PLUGINS/$CONFIG_PLUGINS_FILE
+                        sed -i '$a'\\ "$CONFIG_PLUGINS_FILE"
+                        echo "$VAR_NAME = $VAR_VALUE" >>"$CONFIG_PLUGINS_FILE"
                     fi
                 fi
-            fi 
+            fi
         done
     fi
     # Config template such like {{ platform_etc_dir }}
```

Comment:
I do not understand why we have the config template (not rendered) added to docker.
@zhanghongtong